zio-json/shared/src/main/scala/zio/json/decoder.scala

```diff
@@ -298,6 +315,18 @@ object JsonDecoder extends GeneratedTupleDecoders with DecoderLowPriority0 {
     final case class SumType(cons: String)       extends JsonError
   }
 
+  def manualSumSwitch[A](partialFunction: PartialFunction[Char, JsonDecoder[A]]): JsonDecoder[A] = new JsonDecoder[A] {
```

Comment:
This should be defined in the companion object of `JsonDecoder`. I would suggest calling it `peekChar`.

Reply:
I'm not sure I understand the part about the companion object. It's already in object `JsonDecoder`.
I change the name immediatly.